Сохранение и загрузка

nuclearthinking opened the pull request.

nuclearthinking committed:
тест kryo

nuclearthinking committed:
myheroagency-7 Save and load

nuclearthinking committed:
myheroagency-7 Save and load

Izonami commented on core/src/com/nuclearthinking/myheroagency/model/GameData.java:
```diff
@@ -0,0 +1,52 @@
+    public static void save(GameData gameData) {
+        Kryo kryo = new Kryo();
+        Output output = null;
+        try {
```
Ты же вроде на яву 7 переделывал, она уже поддерживает try с аргементами и Output имплементит Closeable
так что можно переделать как то так 

```
try(Output output = new Output(new FileOutputStream(Constants.SAVE_NAME))) {
            kryo.writeObject(output, gameData);
        } catch (FileNotFoundException ex) {
            new SimpleLoggerFactory().getLogger("GameData").info("Can't save game to file {}", Constants.SAVE_NAME);
        }
```

nuclearthinking replied:
я такой темы не знал

Izonami commented on core/src/com/nuclearthinking/myheroagency/model/GameData.java:
```diff
@@ -0,0 +1,52 @@
+        Kryo kryo = new Kryo();
+        GameData gameData = null;
+        Input input = null;
+        try {
```
и тут по аналогии

```
try(Input input = new Input(new FileInputStream(Constants.SAVE_NAME))) {
            gameData = kryo.readObject(input, GameData.class);
        } catch (FileNotFoundException ex) {
            new SimpleLoggerFactory().getLogger("GameData").info("Can't load game from file", Constants.SAVE_NAME);
        }
```

Izonami commented:
и если есть какой то линк хороший про Kryo кинь плиз, а то я не особо в курсе что это

nuclearthinking committed:
мердж

Izonami commented:
Подливай мастер и комить

nuclearthinking committed:
# Conflicts:
#	core/src/com/nuclearthinking/myheroagency/controller/ScreenEnum.java
#	core/src/com/nuclearthinking/myheroagency/view/AbstractScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/HomeScreen.java

nuclearthinking committed:
исправления

nuclearthinking committed:
# Conflicts:
#	android/assets/font/Kramola.fnt
#	android/assets/font/default.fnt
#	android/assets/i18n/MainMenuScreen.properties
#	android/assets/i18n/MainMenuScreen_en.properties
#	android/assets/i18n/MainMenuScreen_ru.properties
#	android/assets/ui/ui.atlas
#	android/assets/ui/ui.json
#	core/src/com/nuclearthinking/myheroagency/controller/ScreenEnum.java
#	core/src/com/nuclearthinking/myheroagency/view/AbstractScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/HomeScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/MainMenuScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/SettingsScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/SplashScreen.java

nuclearthinking committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into myheroagency-7

# Conflicts:
#	android/assets/font/Kramola.fnt
#	android/assets/font/default.fnt
#	android/assets/i18n/MainMenuScreen.properties
#	android/assets/i18n/MainMenuScreen_en.properties
#	android/assets/i18n/MainMenuScreen_ru.properties
#	android/assets/ui/ui.atlas
#	android/assets/ui/ui.json
#	core/src/com/nuclearthinking/myheroagency/view/AbstractScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/MainMenuScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/SettingsScreen.java
#	core/src/com/nuclearthinking/myheroagency/view/SplashScreen.java

nuclearthinking committed:
гитигнор

nuclearthinking committed:
Merge remote-tracking branch 'origin/myheroagency-7' into myheroagency-7

# Conflicts:
#	.gitignore
